### PR TITLE
[8.0][NEW][crm_autoalias]

### PR DESCRIPTION
**Pull request by hugosantosred**
Add module crm_autoalias to automatically generate email alias in lead/opportunity related to that lead/opportunity.

### CODE REVIEW

File: crm_autoalias/README.rst
Comment (pedrobaeza) [inline crm_autoalias/README.rst:31]: Underscore must have the same length as the title

File: crm_autoalias/README.rst
Comment (pedrobaeza) [inline crm_autoalias/README.rst]: Remove this title

File: crm_autoalias/README.rst
Comment (pedrobaeza) [inline crm_autoalias/README.rst]: What is the expression/sequence to use for the automatic alias? If it's a sequence, put it on the Configuration section what's its name.

File: crm_autoalias/model/crm_lead.py
Comment (pedrobaeza) [inline crm_autoalias/model/crm_lead.py]: Put it more simple: `vals['email_alias'] = alias.id`

File: crm_autoalias/test/test_autoalias.yml
Comment (pedrobaeza) [inline crm_autoalias/test/test_autoalias.yml]: You should test the expected alias

**Comment by pedrobaeza** [comment]
Is there any other module that benefits from this autocreated alias? Does the standard make it something with this?

**Comment by hugosantosred** [comment]
@pedrobaeza This module is intended only to allow to attach email messages to a specific crm.lead by sending an email including in it the alias address of that specific crm.lead.

It's useful for easily attach email conversations about any lead/opportunity by simply forward this emails to the alias address. Odoo itself, with fetchmail configured, automatically attachs this emails to the corresponding lead/opportunity to which the alias belongs.

**Comment by pedrobaeza** [comment]
Please document then in the README what is the purpose of the auto-creation of the alias.

**Comment by pedrobaeza** [comment]
As mentioned by Carlos in the community list, you can link also the link to the tutorial for the catchall configuration (http://pinakinnayi.blogspot.com.es/2013/07/mail-alias-with-openerp-7.html)

**Comment by rafaelbn** [comment]
The functionality to pretend the same as Odoo make with projects. But...

Odoo think that you will have this conversation ("attach email conversations") inside Odoo. You need this module because you create the lead but you talk with the lead out of Odoo, so you are not using Odoo functionality and thats why you need this module.

In my opinion this functionality is the same as in the core. I can make the same of this module without this module:

1 - I sent an email to the lead and followers
2 - Lead receive e-mail and I receive too
3 - I answer this e-mail (catchall@domain.com) and paste on in all conversation and attach all documents
4 - I will get in my lead all conversations about this lead (the same).

"It's useful for easily attach email conversations about any lead/opportunity by simply forward this emails to the alias address."

This is useful if:

1 - You have only a few lead (very few). After a mass mailing campaign or a landing page with form you can have hundreds or thousand of leads.
2 - You use leads only for commercial use (not use for lead created which will be converted in Issues)
3 - You e-mail server is Gmail (not all e-mails servers have autoalias it self. If the company has Exchange, OpenXchange, Zimbra, ... this doesn't work)

"Odoo itself, with fetchmail configured, automatically attachs this emails to the corresponding lead/opportunity to which the alias belongs."

I think Odoo had this functionality on the table but discard it as described above.

@hugosantosred I think you mus use standard functionality for this purpose (it's only my opinion) as it make the same.

File: crm_autoalias/model/crm_lead.py
Comment (hbrunn) [inline crm_autoalias/model/crm_lead.py]: why is this not a stored related field to `['email_alias', 'display_name']`?

File: crm_autoalias/model/crm_lead.py
Comment (hbrunn) [inline crm_autoalias/model/crm_lead.py:16]: add `@api.returns('self', lambda x: x.id)`

File: crm_autoalias/model/crm_lead.py
Comment (hbrunn) [inline crm_autoalias/model/crm_lead.py]: don't duplicate code here. Use `while True:` and break if there's no collision

**Comment by rafaelbn** [comment]
@antoniocanovas maybe you are interested in this PR

**Comment by antoniocanovas** [comment]
Hi Rafa,
I knew this PR but it's not just what I need.

I want to create an "alias" in partner, and it has not to be "autoalias"
(the alias must be easy to remember).
This tool will be a good solution to register an email to de partner
history sent by any other emailing tool, for each person in the company who
knows this alias.

Thank you.

Un saludo,

---

```
   Antonio Cánovas Pedreño
```

antonio.canovas@ingenieriacloud.com
         Teléfono.- 6224.2.6224
-         Twitter.- @acanovasp http://www.twitter.es/acanovasp*

2015-12-10 1:13 GMT+01:00 Rafael Blasco notifications@github.com:

> @antoniocanovas https://github.com/antoniocanovas maybe you are
> interested in this PR
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/crm/pull/27#issuecomment-163443532.

## 

 http://www.konery.es
Antes de imprimir este e-mail piense bien si es necesario, la conservación 
del medio ambiente es tarea de todos.
Este mensaje contiene información CONFIDENCIAL sometida a secreto 
profesional. Si lo ha recibido por error, debe saber que la copia, uso o 
divulgación están prohibidos la Ley, por ello, le rogamos que nos lo 
comunique por esta misma vía y proceda a su destrucción. 

Muchas gracias.

**Comment by pedrobaeza** [comment]
@hugosantosred, can you please attend the PR comments?

**Comment by hugosantosred** [comment]
@pedrobaeza Sorry for the delay, I wan't to attend the coments and fix the code this week.

File: crm_autoalias/model/crm_lead.py
Comment (hbrunn) [inline crm_autoalias/model/crm_lead.py:16]: ignore this comment, the `returns` decorator is inherited

**Comment by hugosantosred** [comment]
@hbrunn Changes done over the code.

Thanks for the review!

File: crm_autoalias/README.rst
Comment (pedrobaeza) [inline crm_autoalias/README.rst:6]: Use latest README template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: crm_autoalias/README.rst
Comment (pedrobaeza) [inline crm_autoalias/README.rst]: Indent this line 2 spaces for a correct formatting

File: crm_autoalias/__openerp__.py
Comment (pedrobaeza) [inline crm_autoalias/__openerp__.py]: Shorten the headers as https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: crm_autoalias/__openerp__.py
Comment (pedrobaeza) [inline crm_autoalias/__openerp__.py]: 8.0.1.0.0

**Comment by pedrobaeza** [comment]
@hugosantosred, you need to include a post-init-hook that creates an alias for all existing leads when installing the module.

**Comment by rafaelbn** [comment]
Hi @hugosantosred are you going to finish this PR ? It not we will close it. Please let me know, thanks

**Comment by hugosantosred** [comment]
@rafaelbn Yes I'll try to finish it this week.

**Comment by hugosantosred** [comment]
The hook was already uploaded in last commit, I tought that this part of code was missing.

Any other changes that could be done in order to merge it to the main branch?

@rafaelbn @pedrobaeza

**Comment by rafaelbn** [comment]
Hi @hugosantosred can you explain me again please why do you need this module?

I use a lot leads/opportunities in Odoo with fulll e-mail integration and I don't understand this module purpose

Thanks

**Comment by hugosantosred** [comment]
@rafaelbn We have a customer that uses e-mail management from outside Odoo, but they want to have the conversations inside their lead/oportunity in Odoo.

Because of this we made an easy option to link any e-mail sent from outside Odoo with the matching lead/opportunity with this module. This works by creating an individual email adress for that lead/opportunity, that can be added as a CC or BCC in the outside e-mail and when is received by odoo the conversation is saved as a message in the lead/opportunity that have this email alias created.

**Comment by rafaelbn** [comment]
So, if you have 50 leads, then you have to:
1 - Go to Odoo -> Sales -> Lead/Opportunity and **one by one**
2 - Copy the alias (**one by one**)
3 - Go to external e-mail management
4 - Create a new email for customer and paste the alias copied in point 2
5 - Then the email goes to the alias in Odoo and conversation is stored
Is like that? After clarification I will tell you my full opinion 😄 
By the moment I see a lot of work for you customer with this mechanism. And about this module, I could be in OCA of course but is not a general purpose, is very specific for your customer.
Appreciated more opinions.

**Review by rafaelbn — APPROVED** [review]
I've tested in runbot, It works and make what is expected to do. My only doubt if is this is necessary, as would be very much simple an fast send emails from Odoo y send them also yo the external email manager.

**Review by pedrobaeza — APPROVED** [review]
OK, I merge anyways as the work is done, for those who find it interesting.

**Pull request merged**
